src/util/color/colorpalette.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <QList>
+
+#include "util/color/rgbcolor.h"
+
+class ColorPalette final {
+  public:
+    explicit ColorPalette(QString name, QList<mixxx::RgbColor> colorList)
+            : m_name(name),
+              m_colorList(colorList) {
+        DEBUG_ASSERT(m_colorList.size() != 0);
```

Comment:
what happens in the release build if this assertion is violated?  if mixxx will crash anyway, better to crash here and just ASSERT

Reply:
ok

Reply:
There is no `ASSERT` and `Q_ASSERT` just prints a message. I'll just leave the debug assert for now.

Reply:
I could use `assert` from `<cassert>`, should I?

Reply:
Q_ASSERT_X exits when false: https://doc.qt.io/qt-5/debug.html#debugging-macros

Reply:
would this crash on startup?  or might it crash if the user switches skins?  It would be nice to front-load this sort of crasher invariant so it's not discovered in the middle of a run

Reply:
We have RELEASE_ASSERT